tests/sentry/tasks/test_find_missing_codemappings.py

```diff
@@ -0,0 +1,295 @@
+                            "lineno": 30,
+                            "filename": "sentry/test_file.py",
+                        },
+                        {
+                            "function": "test_fn_2",
+                            "abs_path": "/usr/src/sentry/src/sentry/models/test_file.py",
+                            "module": "sentry.models.release",
+                            "in_app": True,
+                            "lineno": 39,
+                            "filename": "sentry/models/test_file.py",
+                        },
+                    ]
+                },
+                "fingerprint": ["put-me-in-the-control-group"],
+            },
+            project_id=new_project.id,
+        )
+
+        with self.tasks():
+            mapping = find_missing_codemappings([self.organization, new_org])
+        assert self.organization.slug in mapping
+        result_1 = mapping[self.organization.slug]
+        assert self.project.slug in result_1
+        assert sorted(result_1[self.project.slug]) == [
+            "sentry/models/release.py",
+            "sentry/tasks.py",
+        ]
+        assert new_org.slug in mapping
+        result_2 = mapping[new_org.slug]
+        assert new_project.slug in result_2
+        assert sorted(result_2[new_project.slug]) == [
+            "sentry/models/test_file.py",
+            "sentry/test_file.py",
+        ]
+
+    def test_skips_stale_projects(self):
+        self.store_event(
+            data={
+                "message": "Kaboom!",
+                "platform": "python",
+                "timestamp": iso_format(before_now(days=8)),
+                "stacktrace": {
+                    "frames": [
+                        {
+                            "function": "handle_set_commits",
+                            "abs_path": "/usr/src/sentry/src/sentry/tasks.py",
+                            "module": "sentry.tasks",
+                            "in_app": False,
+                            "lineno": 30,
+                            "filename": "sentry/tasks.py",
+                        },
+                        {
+                            "function": "set_commits",
+                            "abs_path": "/usr/src/sentry/src/sentry/models/release.py",
+                            "module": "sentry.models.release",
+                            "in_app": True,
+                            "lineno": 39,
+                            "filename": "sentry/models/release.py",
+                        },
+                    ]
+                },
+                "fingerprint": ["put-me-in-the-control-group"],
+            },
+            project_id=self.project.id,
+        )
+
+        with self.tasks():
+            mapping = find_missing_codemappings()
+        assert self.organization.slug in mapping
+        result = mapping[self.organization.slug]
+        assert self.project.slug not in result
+
+    def test_handles_duplicates(self):
```

Comment:
```suggestion
    def test_handle_duplicate_frame_filenames_within_same_stacktrace(self):
```
Works for you? It took me a bit to figure out what was duplicated.